### PR TITLE
Compact buttons

### LINKED ISSUE #5839: Compact buttons
### Why this would be helpful

Not so much text

### How this addon works

Only show the icon instead like "(icon)see innside"

### Possible alternatives

I don't think so

### Additional context

I have allready made it :D
(Or a beta version of it)

### PR DESCRIPTION
**Pull request by StioStudio**
Resolves #5839
(That it is to much text on the editor and the project page)

### Changes
*Hides all the text on the buttons: turbowarp, see inside, share, remix
*adds a icon to the share button

### Reason for changes
I don't like that it is so much text
I made this mostly for my self, but I thought I could send a pr

### Tests
Tested on oprea GX. It is fully CSS and I have not used any weird code.

### CODE REVIEW

**Comment by StioStudio** [comment]
I forgot to take away the info page. Just take that away if you accept the pr :)

**Comment by BroJac5246** [comment]
> I forgot to take away the info page. Just take that away if you accept the pr :)

Please do it yourself or open a new PR. The removal of it would need to be reviewed, so do it now so we don't have to wait even longer later.

Also, in general you should find or create an issue before making a PR.

**Comment by StioStudio** [comment]
> > I forgot to take away the info page. Just take that away if you accept the pr :)
> 
> Please do it yourself or open a new PR. The removal of it would need to be reviewed, so do it now so we don't have to wait even longer later.
> 
> Also, in general you should find or create an issue before making a PR.

Yeah sorry about that, but how do I hide the other addon from the pr?

**Comment by BroJac5246** [comment]
> > > I forgot to take away the info page. Just take that away if you accept the pr :)
> > 
> > Please do it yourself or open a new PR. The removal of it would need to be reviewed, so do it now so we don't have to wait even longer later.
> > 
> > Also, in general you should find or create an issue before making a PR.
> 
> Yeah sorry about that, but how do I hide the other addon from the pr?

Delete all the code from it. If you still want the code, then back it to a different branch.

**Comment by StioStudio** [comment]
> > > > I forgot to take away the info page. Just take that away if you accept the pr :)
> > > 
> > > 
> > > Please do it yourself or open a new PR. The removal of it would need to be reviewed, so do it now so we don't have to wait even longer later.
> > > Also, in general you should find or create an issue before making a PR.
> > 
> > 
> > Yeah sorry about that, but how do I hide the other addon from the pr?
> 
> Delete all the code from it. If you still want the code, then back it to a different branch.

Ok I will see what I can do (I am not good with git and GitHub)

**Comment by DNin01** [comment]
> > > > > I forgot to take away the info page. Just take that away if you accept the pr :)
> 
> > > > 
> 
> > > > 
> 
> > > > Please do it yourself or open a new PR. The removal of it would need to be reviewed, so do it now so we don't have to wait even longer later.
> 
> > > > Also, in general you should find or create an issue before making a PR.
> 
> > > 
> 
> > > 
> 
> > > Yeah sorry about that, but how do I hide the other addon from the pr?
> 
> > 
> 
> > Delete all the code from it. If you still want the code, then back it to a different branch.
> 
> 
> 
> Ok I will see what I can do (I am not good with git and GitHub)

You can re-commit everything related to the compact buttons feature to a new branch in order to remove the extra unrelated commits, then create a new PR.

Be sure to include a link to the issue you created for this by typing `Resolves #` followed by the issue number.

**Comment by StioStudio** [comment]
> > > > I forgot to take away the info page. Just take that away if you accept the pr :)
> > > 
> > > 
> > > Please do it yourself or open a new PR. The removal of it would need to be reviewed, so do it now so we don't have to wait even longer later.
> > > Also, in general you should find or create an issue before making a PR.
> > 
> > 
> > Yeah sorry about that, but how do I hide the other addon from the pr?
> 
> Delete all the code from it. If you still want the code, then back it to a different branch.

I fixed it

**Comment by StioStudio** [comment]
@DNin01 thanks for the change in the "Resolves"

**Comment by StioStudio** [comment]
Should I take away the "beta" tag

**Comment by StioStudio** [comment]
And what should I set the "versionAdded" to? Should it be like 1.31.1 or maybe 1.31.2?

File: addons/compact-buttons/addon.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "name": "Compact Buttons",
```
Comment (DNin01) [inline addons/compact-buttons/addon.json:2]: We don't usually use Start Case in addon names as of now.
```suggestion
    "name": "Compact buttons",
```

File: addons/compact-buttons/addon.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "description": "Buttons in projects get smaller/more compact",
```
Comment (DNin01) [inline addons/compact-buttons/addon.json:3]: Might want to further clarify this.
```suggestion
    "description": "Turns the text on some buttons on project and editor pages into icons to save space.",
```

**Comment by StioStudio** [comment]
So is this gonna be accepted?

**Comment by StioStudio** [comment]
Hello?

**Comment by Samq64** [comment]
> So is this gonna be accepted?

Probably, but it might be a while before this is reviewed.

**Comment by StioStudio** [comment]
> > So is this gonna be accepted?
> 
> Probably, but it might be a while before this is reviewed.

Oh ok

**Comment by WorldLanguages** [comment]
So this also involves creating our own icons for existing buttons that lack an icon, right?

**Comment by StioStudio** [comment]
> So this also involves creating our own icons for existing buttons that lack an icon, right?

Yes. For now that is only the share button that I made a icon for (made it in like 2 minutes, so it is not that good).

**Comment by WorldLanguages** [comment]
@StioStudio That makes this a bit more complex, because we need to find suitable icons that can be understood 🤔

**Comment by StioStudio** [comment]
> @StioStudio That makes this a bit more complex, because we need to find suitable icons that can be understood 🤔

My icons works pretty good

**Comment by StioStudio** [comment]
> @StioStudio That makes this a bit more complex, because we need to find suitable icons that can be understood 🤔

Btw if you want to see the icon: https://github.com/ScratchAddons/ScratchAddons/pull/5840/files#diff-4677ea56a0fb5544a11b4475a31d26ee65d6f3717437d66f4e28f5d151ac00e9

**Comment by Secret-chest** [comment]
What about a separate toggle for each button type?

**Comment by StioStudio** [comment]
> What about a separate toggle for each button type?

Ummm.. I REALLY don't want to do that, but sure. It is a good idea.

File: addons/compact-buttons/banner-button.css [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    height: 1.125rem;
+    vertical-align: middle;
+    content: "";
+}
+.sa-unshare-button .button_content_3jdgj {
+    background-image: url(unshare.svg);
+}
+.share-button_share-button_Nxxf0 .button_content_3jdgj span {
+    display: none;
+}
```
Comment (iqnite) [inline addons/compact-buttons/banner-button.css:51]: Add a newline at the end of every file in order to make the Prettier checks succeed:
```suggestion
}

```

File: addons/compact-buttons/addon.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        "matches": ["projects"]
+      },
+      {
+        "url": "see-inside-button.css",
+        "matches": ["projects"]
+      }
+    ],
+    "tags": ["community"],
+    "versionAdded": "1.32.0",
+    "enabledByDefault": false
```
Comment (iqnite) [inline addons/compact-buttons/addon.json:28]: Adding the ability to toggle the addon without having to reload the page would make the whole thing easier for many people:
```suggestion
    "enabledByDefault": false,
    "dynamicEnable": true,
    "dynamicDisable": true
```

**Review by iqnite — CHANGES_REQUESTED** [review]
The addon itself should work (didn't test right now), but why did you make a different CSS file for every button? Can't you combine many different items in one CSS declaration?

**Comment by StioStudio** [comment]
> The addon itself should work (didn't test right now), but why did you make a different CSS file for every button? Can't you combine many different items in one CSS declaration?

Because the file would be really long and so I have more overview over the CSS. And maybe because I forgot how to comments XD
I can fix it though :)

**Comment by BroJac5246** [comment]
> > The addon itself should work (didn't test right now), but why did you make a different CSS file for every button? Can't you combine many different items in one CSS declaration?
> 
> Because the file would be really long and so I have more overview over the CSS. And maybe because I forgot how to comments XD
> I can fix it though :)

CSS comments use the `/* */` format that JavaScript multi-line comments use. They can be single- or multi-line. [Source](https://www.w3schools.com/css/css_comments.asp)

**Comment by Secret-chest** [comment]
If you use a separate file for each one, you can add settings

**Comment by StioStudio** [comment]
Hey. I was wondering if this could be reviewed soon.

**Comment by StioStudio** [comment]
Hi @WorldLanguages . Sorry for the ping but I was wondering if this could be reviewed/accepted soon.

**Comment by WorldLanguages** [comment]
I believe there hasn't been enough discussion regarding how we'd decide which icons to use for buttons that don't have icons, such as share/unshare - and whether we want to offer that functionality in the first place.

**Comment by WorldLanguages** [comment]
I also wonder how we would deal with the inconsistency of removing the words "see inside" from the project page, but keeping "see project page" on the editor. If we decide to let this addon handle both, this would set a weird precedent as the same addon would be changing UI in the website and editor at the same time 🤔

**Comment by WorldLanguages** [comment]
I also would like to know if there's any specific reason the "compact buttons" concept is being restricted just to project pages? Couldn't it also be extended to profiles, studios, and other pages?

**Comment by Samq64** [comment]
> I also wonder how we would deal with the inconsistency of removing the words "see inside" from the project page, but keeping "see project page" on the editor.

I think that could be a toggle in #4018 along with #5866.

**Comment by StioStudio** [comment]
> I believe there hasn't been enough discussion regarding how we'd decide which icons to use for buttons that don't have icons, such as share/unshare - and whether we want to offer that functionality in the first place.

I allready have made the icons and I have used the addon the last 2 weeks. They work pretty good allready (the ones I made)

**Comment by StioStudio** [comment]
> I also wonder how we would deal with the inconsistency of removing the words "see inside" from the project page, but keeping "see project page" on the editor. If we decide to let this addon handle both, this would set a weird precedent as the same addon would be changing UI in the website and editor at the same time 🤔

I don't totaly understand this. The text goes away on editor and page. I don't see the problem.

**Comment by StioStudio** [comment]
> I also would like to know if there's any specific reason the "compact buttons" concept is being restricted just to project pages? Couldn't it also be extended to profiles, studios, and other pages?

That is just because it is what I made. I was thinking to expand it to other things too.

**Comment by Secret-chest** [comment]
I would really like a way to control each type of button separately.

**Comment by StioStudio** [comment]
> I would really like a way to control each type of button separately.

Yeah. I need to learn SA settings first though

**Comment by BroJac5246** [comment]
> > I would really like a way to control each type of button separately.
> 
> Yeah. I need to learn SA settings first though

[addon.settings docs](https://scratchaddons.com/docs/reference/addon-api/addon.settings/)

**Comment by StioStudio** [comment]
Don't mind the commits. It was just me doing it the wrong way

**Comment by TylerMS887** [comment]
Can I have a screenshot of this (or the Chrome extension file)?

**Comment by StioStudio** [comment]
> Can I have a screenshot of this (or the Chrome extension file)?

What do you mean?

**Comment by TylerMS887** [comment]
> > Can I have a screenshot of this (or the Chrome extension file)?
> 
> What do you mean?

If you were talking about the screenshot, I want to see what this addon looks like. Click the image button on the top of the editor to upload a file to GitHub and insert it.
If you use Windows 11 Insider, open the Snipping Tool and select the modified parts.

If you were talking about the extension file, the Chrome extension file is a file that can be imported into Chromium based browsers in developer mode. In this case, the `packer.yml` file is missing so you will need to build yourself.

**Comment by StioStudio** [comment]
image
image

**Comment by StioStudio** [comment]
> > > Can I have a screenshot of this (or the Chrome extension file)?
> > 
> > 
> > What do you mean?
> 
> If you were talking about the screenshot, I want to see what this addon looks like. Click the image button on the top of the editor to upload a file to GitHub and insert it. If you use Windows 11 Insider, open the Snipping Tool and select the modified parts.
> 
> If you were talking about the extension file, the Chrome extension file is a file that can be imported into Chromium based browsers in developer mode. In this case, the `packer.yml` workflow file is missing from your fork so you will need to build yourself.

Does that work?

**Comment by Secret-chest** [comment]
Can it get settings for each button type? I want TurboWarp and Remix to not have icons but See inside and Share to have one.

**Pull request closed without merging**